windows/System/File/Platform.hsc

```diff
@@ -43,11 +43,11 @@ openFile fp iomode = bracketOnError
     toHandle
  where
 #if defined(__IO_MANAGER_WINIO__)
-  toHandle h = do
+  toHandle h = (`onException` Win32.closeHandle h) $ do
```

Comment:
Not 100% sure if we want `closeHandle` here, but I can't see any such attempt in `hANDLEToHandle` etc. So I guess we do that here.